### PR TITLE
Add Roadmaps

### PR DESCRIPTION
**Pull request by liuchong**
Add https://github.com/liuchong/awesome-roadmap

I've just made a collection of roadmaps, will add more in the following days.

Also I've found an issue https://github.com/sindresorhus/awesome/issues/1470 refer to this, so I'd like to share my list here.

### CODE REVIEW

**Comment by troxler** [comment]
It does not seem you have read the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) at all.

**Comment by ugurozturk** [comment]
@liuchong 

> Requirements for your Awesome list
> Has been around for at least 30 days.
> That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.

**Comment by liuchong** [comment]
Thanks, I've just made some updates on README.md to fit the contribution guidelines format, and will improve the list in the waiting 30 days :joy_cat:

**Comment by damccorm** [comment]
Outside of the waiting period, this looks like it now conforms to the style guide and is a cool list. Once its been around 30 days (aka early February) I'd vote to merge this in.

**Comment by alexk111** [comment]
Thanks for the list!

> Has been around for at least 30 days.

It was already mentioned :)

> Has contribution guidelines. People who are contributing to your list should have a clear understanding of how they should do so.

contributing.md doesn't have any contribution guidelines.

> Consistent and correct naming. For example, Node.js, not NodeJS or node.js

Double-check this. Vue is Vue.js: https://vuejs.org/

**Comment by liuchong** [comment]
@alexk111 Thanks for pointing out the problems!
I've just filled `contributing.md`, and fixed name of `VUE.JS` in item description, but did not change the link, as it is exactly the project name.

**Comment by alexk111** [comment]
I'm not sure that `VUE.JS` is correct. I believe the correct one is `Vue.js`. Check the top-left corner at https://vuejs.org.

**Comment by alexk111** [comment]
This is a style for all the buttons in the center. Just checked [their repo](https://github.com/vuejs/vue) - the correct one is definitely `Vue.js`.

Thanks for the corrections! 👍

**Comment by liuchong** [comment]
30 days passed, I have rebased my pull request branch to slove conflicts, could you check it again to see if it acceptable now?

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/liuchong/awesome-roadmap
✖ Linting

  README.md:9:3
  ✖   1:20   Invalid badge source                                    remark-lint:awesome/badge
  ✖   9:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  10:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  11:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  12:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  13:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  14:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  15:3    Incorrect indentation before bullet: remove 2 spaces    remark-lint:list-item-bullet-indent
  ✖  18:121  List item description must end with proper punctuation  remark-lint:awesome/list-item
  ✖  19:132  List item description must end with proper punctuation  remark-lint:awesome/list-item
  ✖  21:153  List item description must end with proper punctuation  remark-lint:awesome/list-item
  ✖  39:17   Remove empty section: "Miscellaneous"                   remark-lint:no-empty-sections

  12 errors
```

https://github.com/sindresorhus/awesome-lint

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by sindresorhus** [comment]
> This pull request has a descriptive title in the format Add Name of List (Example: Add Swift), not Update readme.md or Add awesome list.

**Comment by sindresorhus** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)

File: readme.md [hunk 1]
```diff
@@ -605,7 +605,7 @@
 - [Speakers](https://github.com/karlhorky/awesome-speakers#readme) - Conference and meetup speakers in the programming and design community.
 - [Software Patreons](https://github.com/uraimo/awesome-software-patreons#readme) - Fund individual programmers or the development of open source projects.
 - [Parasite](https://github.com/ecohealthalliance/awesome-parasite#readme) - Parasites and host-pathogen interactions.
-
+- [Roadmap](https://github.com/liuchong/awesome-roadmap#readme) - A curated list of roadmaps.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
Include a short description about the project/theme of the list. **It should not describe the list itself.**

**Comment by liuchong** [comment]
Ok, I have fixed the lint errors and merge confilcts, please check again, thanks.

**Comment by liuchong** [comment]
@sindresorhus rebased again :joy_cat:

**Comment by MoienTajik** [comment]
**Ping!** Why this is not merged yet? I think this is a useful one.

**Comment by liuchong** [comment]
Hi @sindresorhus , several months passed, would this ok to merge?

**Comment by sindresorhus** [comment]
`🚀 iOS Developer Roadmap` => `iOS Developer Roadmap`

`To give you an overview of the core skills needed in data science.` => `Gives you an overview of the core skills needed in data science.`

File: readme.md [hunk 2]
```diff
@@ -435,6 +435,7 @@
 - [Educational Games](https://github.com/yrgo/awesome-eg#readme) - Learn while playing.
 - [JavaScript Learning](https://github.com/micromata/awesome-javascript-learning#readme)
 - [CSS Learning](https://github.com/micromata/awesome-css-learning#readme) - Mainly about CSS – the language and the modules.
+- [Roadmaps](https://github.com/liuchong/awesome-roadmaps#readme) - Give you a clear route to improve your knowledge or skills.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
- [Roadmaps](https://github.com/liuchong/awesome-roadmaps#readme) - Gives you a clear route to improve your knowledge and skills.
```

**Comment by liuchong** [comment]
Thanks for the thoroughly check, the issue are fixed now 💯

File: readme.md (hunk 2 above)
Comment (liuchong) [reply readme.md]: Thanks!

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1127792300933795840

**Comment by liuchong** [comment]
> Tweet: https://twitter.com/awesome__re/status/1127792300933795840

Thanks!

**Pull request merged**